PigLatin/Program.cs

```diff
@@ -77,16 +80,17 @@ public static string TranslateWord(string word)
                 Console.WriteLine(word + " Translated by word has no vowels");
                 return word = string.Concat(word, addAy);
             }
-        }
+        } 
     public static string TranslatePunctuation(string punctuation)
 {
     var punctuationIndex = punctuation.IndexOfAny(new char[] { '.', ',', '!', '?' });
     string[] allPunctuation = punctuation.Split('.', ',', '!', '?');
```

Comment:
You are getting close here. Not all works will have punctuation. Only the last work of a sentence. So try splitting the word, then check if each word has a punctuation at the end of it.